portfolio/src/main/webapp/script.js

```diff
@@ -81,29 +82,78 @@ function createCommentElement(comment) {
   const commentInputElement = document.createElement('p');
   commentInputElement.innerText = comment.commentInput;
 
+  const deleteButtonElement = document.createElement('button');
+  deleteButtonElement.innerText = 'Delete';
+  deleteButtonElement.addEventListener('click', () => {
+    deleteComment(comment);
+    commentBlock.remove();
+  });
+
   commentBlock.appendChild(nameElement);
   commentBlock.appendChild(commentInputElement);
+  commentBlock.appendChild(deleteButtonElement);
   commentBlock.appendChild(document.createElement('hr'));
   return commentBlock;
 }
 
-/*
+/** Tells the server to delete the comment. */
+deleteComment = async (comment) => {
+  const params = new URLSearchParams();
+  params.append('id', comment.id);
+  let response = await fetch('/delete-data', {method: 'POST', body: params});
+  getBlogComments(window.localStorage.getItem("comment-limit"));
```

Comment:
From line 142, you have this setting the dfault comment-limit:

```
let commentLimit = window.localStorage.getItem("comment-limit");
if (!commentLimit) commentLimit = 5;
```

just ooc what the reason is with setting the default in the `window.onload` but not in the `deleteComment`?

Reply:
I thought about it in deleteComment but figured that it wouldn't matter because the importance of setting the default is to make sure that commentLimit has a value - in case the user didn't set a value on the website, so the fetching of comments doesn't fail. (does that make sense?)

Reply:
isnt it possible that with the `deleteComment`,  the user also didn't set a value on the website, in which case we want some default value to be added into the `getBlogComments`?

or am i misunderstanding some of the flow here?